<launch>


	<!--============================= Airsim ros noed============================= -->

	<arg name="output" default="screen"/>
	<arg name="publish_clock" default="false"/>
	<arg name="is_vulkan" default="true"/>
	<arg name="host" default="localhost" />
	<arg name="frame" default="/world_enu"/>

	<node name="airsim_node" pkg="airsim_ros_pkgs" type="airsim_node" output="$(arg output)">
		<param name="is_vulkan" type="bool" value="false" /> 
		<!-- ROS timer rates. Note that timer callback will be processed at maximum possible rate, upperbounded by the following ROS params -->
		<param name="update_airsim_img_response_every_n_sec" type="double" value="0.05" /> 
		<param name="update_airsim_control_every_n_sec" type="double" value="0.01" />
		<param name="update_lidar_every_n_sec" type="double" value="0.01" />
		<param name="publish_clock" type="bool" value="$(arg publish_clock)" />
		<param name="host_ip" type="string" value="$(arg host)" />
		<param name="odom_frame_id" type="string" value="odom_local_enu" />
		<param name="world_frame_id" type="string" value="$(arg frame)" />
	
	</node>
	<!-- Static transforms -->
	<include file="$(find airsim_ros_pkgs)/launch/static_transforms.launch"/>
	<!--============================= Airsim ros noed============================= -->


    <node name="easysim_ros_wrapper" pkg="easysim_ros_wrapper" type="easysim_ros_node" output="screen">
        <param name="ip_address" value="127.0.0.1"/>
        <param name="port" value="50502"/>
        <param name="timer_duration" value="0.05"/>
        <param name="coordinate_frame" value="world_enu"/>
        <param name="use_noise" value="true" />

        <param name="noise_x_std_dev" value="5.0" />
        <param name="noise_y_std_dev" value="5.0" />
        <param name="noise_z_std_dev" value="0.5" />

    </node>


    <node pkg="easysim_ros_wrapper" type="image_ros_wrapper" name="image_ros_wrapper" output="screen">
    </node>


    <node name="se3_controller_node" pkg="se3controller" type="se3controller_ros_node" output="screen">
       
        <param name="Kp_x" value="2" />
        <param name="Kp_y" value="2" />
        <param name="Kp_z" value="0.85" />
        <param name="Kv_x" value="2" />
        <param name="Kv_y" value="2" />
        <param name="Kv_z" value="0.85" />
        <param name="attctrl_tau" value="0.35" />
        <param name="norm_thrust_const" value="0.01725" />
        <param name="max_fb_acc" value="50" />  
        <param name="take_off_height" value="8" />

        <remap from="odometry_topic" to="/airsim_node/drone_1/odom_local_enu" />
        <remap from="position_with_yaw" to="/command/position_target" />

    </node>

    <node name="traj_server" pkg="traj_server" type="traj_server" output="screen" >
        <remap from="input_trajectory_topic" to="/command/trajectory" />
        <remap from="cmd2controller" to="command/position_target" />
    </node>


</launch>


    <!-- <node name="se3_controller_node" pkg="se3controller" type="se3controller_ros_node" output="screen">
        <param name="Kp_x" value="3" />
        <param name="Kp_y" value="3" />
        <param name="Kp_z" value="2.2" />
        <param name="Kv_x" value="2" />
        <param name="Kv_y" value="2" />
        <param name="Kv_z" value="1.5" />
        <param name="attctrl_tau" value="0.35" />
        <param name="norm_thrust_const" value="0.0605" />
        <param name="max_fb_acc" value="30" />  
        <param name="take_off_height" value="8" />

        <remap from="odometry_topic" to="/airsim_node/drone_1/odom_local_enu" />
        <remap from="position_with_yaw" to="/command/position_target" />

    </node> -->
